optashift-employee-rostering-shared/src/main/java/org/optaplanner/openshift/employeerostering/shared/employee/EmployeeAvailability.java

```diff
@@ -38,56 +37,56 @@
 public class EmployeeAvailability extends AbstractPersistable {
 
     @NotNull
     @ManyToOne(fetch = FetchType.EAGER)
     private Employee employee;
 
     @NotNull
-    private LocalDate date; // TODO remove me
+    private OffsetDateTime startDateTime;
     @NotNull
-    private OffsetTime startTime; // TODO change into OffsetDateTime startDateTime
-    @NotNull
-    private OffsetTime endTime; // TODO change into OffsetDateTime endDateTime
+    private OffsetDateTime endDateTime;
 
     @NotNull
     private EmployeeAvailabilityState state;
 
     @SuppressWarnings("unused")
     public EmployeeAvailability() {}
 
-//    public EmployeeAvailability(Integer tenantId, Employee employee, OffsetDateTime startDateTime, OffsetDateTime endDateTime, EmployeeAvailabilityState state) {
-//        this(tenantId, employee, startDateTime.toLocalDate(), startDateTime.toOffsetTime(), endDateTime.toOffsetTime());
-//        this.state = state;
-//    }
-
-    // TODO use OffsetDateTime startDateTime and OffsetDateTime endDateTime instead, remove this constructor
-    public EmployeeAvailability(Integer tenantId, Employee employee, LocalDate date, OffsetTime startTime, OffsetTime endTime) {
+    public EmployeeAvailability(Integer tenantId, Employee employee, OffsetDateTime startDateTime, OffsetDateTime endDateTime) {
         super(tenantId);
         this.employee = employee;
-        this.date = date;
-        this.startTime = startTime;
-        this.endTime = endTime;
+        this.startDateTime = startDateTime;
+        this.endDateTime = endDateTime;
+        if (getDuration().getSeconds() / (60 * 60) >= 28) {
+            throw new IllegalArgumentException("Duration between startDateTime [" + startDateTime + "] and " +
+                    "endDateTime [" + endDateTime + "] greater than 28 hours.");
+        }
     }
 
-    // TODO use OffsetDateTime startDateTime and OffsetDateTime endDateTime instead, remove this constructor
-    public EmployeeAvailability(EmployeeAvailabilityView employeeAvailabilityView, Employee employee, LocalDate date, OffsetTime startTime, OffsetTime endTime) {
+    public EmployeeAvailability(EmployeeAvailabilityView employeeAvailabilityView, Employee employee) {
         super(employeeAvailabilityView);
         this.employee = employee;
-        this.date = date;
-        this.startTime = startTime;
-        this.endTime = endTime;
+        this.startDateTime = employeeAvailabilityView.getStartDateTime();
+        this.endDateTime = employeeAvailabilityView.getEndDateTime();
+        if (getDuration().getSeconds() / (60 * 60) >= 28) {
+            throw new IllegalArgumentException("Duration between startDateTime [" + startDateTime + "] and " +
+                    "endDateTime [" + endDateTime + "] greater than 28 hours.");
+        }
+
     }
 
-    // add method getDuration() and validate it's less than 28 hours (DST buffer)
```

Comment:
Although it's checked in the constructor, the duration can become 30 hours or more by calling the setEndDateTime(), after construction.

Instead of checking it in the constructor, add a @Valid javax.validation annotation to check if the duration is max 28.
More ideal would be to use the @Max annotation, but that's only possible with javax.validation 2.0, which is probably not in WildFly 11 yet:
  http://www.baeldung.com/javax-validation